server/src/main/java/org/apache/cassandra/sidecar/handlers/CompactionStopHandler.java

```diff
@@ -0,0 +1,168 @@
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+import org.apache.cassandra.sidecar.utils.CassandraInputValidator;
+import org.apache.cassandra.sidecar.utils.InstanceMetadataFetcher;
+import org.jetbrains.annotations.NotNull;
+
+import static org.apache.cassandra.sidecar.utils.HttpExceptions.wrapHttpException;
+
+/**
+ * Handler for stopping compaction operations via the Cassandra Compaction Manager.
+ *
+ * <p>Handles {@code POST /api/v1/cassandra/operations/compaction/stop} requests to stop
+ * compaction operations. Expects a JSON payload with compaction_type and/or compaction_id:
+ * <pre>
+ *   { "compaction_type": "COMPACTION", "compaction_id": "abc-123" }
+ * </pre>
+ */
+public class CompactionStopHandler extends AbstractHandler<CompactionStopRequestPayload> implements AccessProtected
+{
+    /**
+     * Constructs a handler with the provided {@code metadataFetcher}
+     *
+     * @param metadataFetcher the metadata fetcher
+     * @param executorPools   executor pools for blocking executions
+     * @param validator       validator for Cassandra-specific input
+     */
+    @Inject
+    protected CompactionStopHandler(final InstanceMetadataFetcher metadataFetcher,
+                                    final ExecutorPools executorPools,
+                                    final CassandraInputValidator validator) {
+        super(metadataFetcher, executorPools, validator);
+    }
+
+    @Override
+    public Set<Authorization> requiredAuthorizations() {
+        return Collections.singleton(BasicPermissions.MODIFY_COMPACTION.toAuthorization());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleInternal(RoutingContext context,
+                               HttpServerRequest httpRequest,
+                               @NotNull String host,
+                               SocketAddress remoteAddress,
+                               CompactionStopRequestPayload request) {
+        CompactionManagerOperations compactionManagerOps = metadataFetcher.delegate(host).compactionManagerOperations();
+
+        executorPools.service()
+                .executeBlocking(() -> stopCompaction(compactionManagerOps, request))
+                .onSuccess(context::json)
+                .onFailure(cause -> processFailure(cause, context, host, remoteAddress, request));
+    }
+
+    /**
+     * Stops the compaction based on the request parameters
+     *
+     * @param operations the compaction manager operations
+     * @param request    the request payload containing compaction_type and/or compaction_id
+     * @return CompactionStopResponse with the operation result
+     */
+    private CompactionStopResponse stopCompaction(CompactionManagerOperations operations,
+                                                  CompactionStopRequestPayload request)
+    {
+        CompactionType compactionType = request.compactionType();
+        String compactionId = request.compactionId();
```

Comment:
Do you want to trim the compactionID? I see that you are doing it in the other parts of the code